### PR TITLE
Fix: invalid

### PR DESCRIPTION
**Pull request by hhugo**

### CODE REVIEW

File: test/passing/attributes.ml [hunk 1]
```diff
@@ -70,7 +70,7 @@ if [@test] true then () else ()
 ;;
 if [@test] true then () else if [@test] true then () else ()
 
-let _ = (A [@test], (() [@test]), ([] [@test]), [||] [@test])
+let _ = ((A [@test]), (() [@test]), ([] [@test]), [||] [@test])
```
[the remark this answers is not in the document]
Comment (hhugo) [reply test/passing/attributes.ml]: it is not intended.

File: src/Ast.ml [hunk 2]
```diff
@@ -1875,6 +1875,10 @@ end = struct
             Pexp_apply ({pexp_desc= Pexp_ident {txt= Lident "not"}}, _) } )
       when is_infix op && not (e1 == exp) ->
         true
+    | ( Exp {pexp_desc= Pexp_apply (e, _)}
+      , {pexp_desc= Pexp_construct _ | Pexp_variant _} )
+      when e == exp ->
+        true
```
[the remark this answers is not in the document]
Comment (hhugo) [reply src/Ast.ml]: The issue was spotted while trying to reformat this file
https://bitbucket.org/camlspotter/ppx_curried_constr/src/8b1c5ab26b12ce9eda3a2bf8b407bd40cc6abcc0/tests/test.ml?at=default&fileviewer=file-view-default

[the remark this answers is not in the document]
Comment (jberdine) [reply src/Ast.ml]: Ah, I see, this is for formatting bogus code that is fixed by a ppx. That would explain that maybe it doesn't fit into the precedence system.

File: test/passing/attributes.ml (hunk 1 above)
[the remark this answers is not in the document]
Comment (jberdine) [reply test/passing/attributes.ml]: Looks like the precedence code has been asking for parens here already and we just didn't emit them.

**Pull request merged**